scripts/check_icons.sh

```diff
@@ -66,32 +105,85 @@ check_is_not_boilerplate_icon() (
 get_icon_from_makefile() (
     repo="$1"
     device_name="$2"
+
+    # Get the Makefile that contains the ICONNAME definitions, copy it and remove its includes and logs
     iconname_makefile=$(grep -Rl --include="*Makefile*" "^[[:blank:]]*ICONNAME" "$repo")
-    grep "ICONNAME" "$iconname_makefile" | grep "$device_name" | cut -f2 -d'=' | sed 's/^[ \t]*//;s/[ \t]*$//'
+    if [[ -z "$iconname_makefile" ]]; then
+        >&2 echo "ERROR => No Makefile with ICONNAME definition found"
+        return 1
+    fi
+    tmp_makefile="/tmp/iconname_makefile.mk"
+    cp "$iconname_makefile" "$tmp_makefile"
+    sed -i "/^include/d" "$tmp_makefile"
+    sed -i "/^\$(info/d" "$tmp_makefile"
+    sed -i "/^\$(error/d" "$tmp_makefile"
+
+    echo "include $tmp_makefile" > "$repo/Makefile_dumper.mk"
+    echo "dump_ICONNAME:" >> "$repo/Makefile_dumper.mk"
+    echo -e "\t@echo \$(ICONNAME)" >> "$repo/Makefile_dumper.mk"
+
+    target_name="TARGET_${device_name^^}"
+    icon=$(make BOLOS_SDK="none" TARGET_NAME="$target_name" --no-print-directory -C "$repo" -f "Makefile_dumper.mk" dump_ICONNAME)
```

Comment:
This doesn't handle icon for variants if any. But I guess this is a great first step

Reply:
True. TBH I would like to export the constraint that icons should be placed in `app-xyz/icons` and be done with this mess checking